tests/load_test/README.md

```diff
@@ -0,0 +1,84 @@
+
+To better measure the stability and find the bottle neck here, this test takes advantage of [Locust](https://locust.io/), which is a high performance open source load testing tool. It simulates up to millions of simultaneous users sending notifications to the target Lira, plots the results in real time, and optionally saves the result to CSV files.
+
+## Usage
+
+This test suite has the following file structure:
+```
+├── README.md
+├── data/  # This folder serves all data for performing a load test
+│   ├── lira_config/  # This folder serves the template file of listener config
+│   │   └── load_test_lira_config.json.ctmpl
+│   ├── plots/  # This folder is used to store metrics plots
+│   ├── notification/  # This folder serves the notification file for the test
+│   │   └── notification.json
+│   ├── results/  # This folder is used to store metrics result JSON files
+│   └── wdl/  # This folder serves the Mock Workflow for the test, which will be used by the Lira directly via Github
+├── load_test/  # This folder serves the models of the test suite
+│   ├── __init__.py
+│   └── models.py
+├── locust_test.py  
+├── normal_loadtest.py  # This is the normal test agent
+├── render_lira_config.sh  # This is used to render the Lira config
+├── requirements.txt  # This file defines all dependencies of this load test
+├── run_locust_test.sh  # This is the script to run the locsut test agent
+└── visualizer.R  # This util is used to plot the metrics
+```
+
+To perform a load test, before all, you need to prepare the data and configs as follows:
+1. Make sure you have the vault token and then render the Lira config by `bash render_lira_config.sh ${ENV} ${DRY_RUN} ${VAULT_TOKEN_FILE}`.
+2. Make sure you are using Python3, then install the dependencies by `pip3 install -r requirements.txt`
+
+### Normal Testing Agent
+To use the normal testing agent, you need to pass in the following parameters:
+```bash
+python normal_loadtest.py \
+    --counts=10 \
+    --url='http://pipelines.dev.data.humancellatlas.org/notifications' \
+    --auth_token=${TOKEN_VALUE} \
+    --payload_file='./data/notification/notification.json' \
+    --result_folder='./data/' \
+    --lira_version='v0.7.0' \
+    --environment='gke' \
+    --mode='dry_run' \
+    --cromwell_type='instance' \
+    --caching='True'
+```
+*Note*: the last 5 parameters are non-functional, they are used to generate the metadata of the result metrics.
+
+### Locust Distributed Testing Agent
+To use the distributed testing agent, you need to modify the `content`, `params`, `notification_ratio` sections of the locust.json file in this directory. 
+- content contains JSON payload you will send to the targeting host.
+- params is the auth-token-value of `{"auth": auth-token-value}` to be sent with the POST requests. 
+- notication_ratio is non-functional at this time.
```

Comment:
Based on https://docs.locust.io/en/stable/api.html#task-decorator, the notification_ratio is a rough "weight" instead of the actual number of notifications each user will send to the server. There isn't a way to control the number of requests each user will send to the server. Besides, if there are two tasks: task1(1) and task2(100), the ratio will work and execute tasks at a 1:100 rate. But if there is only one weight, like what we have here, it won't properly work.

Reply:
Ok, thanks for the clarification!